credentials/credentials.go

```diff
@@ -84,9 +92,38 @@ func (c *Credentials) Decrypt(input io.Reader) (io.Reader, error) {
 	return out, nil
 }
 
+func (c *Credentials) EncryptV2(data []byte) ([]byte, error) {
```

Comment:
Is this used outside of tests? I don't think clients need to be able to encrypt stuff, so I would probably move this to a file with a `tests` build tag so it's only included in the test binary.